server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -523,8 +523,25 @@ export default class ServiceProviderReferralsController {
       formError = form.error
 
       if (form.isValid) {
-        await this.interventionsService.updateDraftActionPlan(token, actionPlanId, form.paramsForUpdate)
-        return res.redirect(`/service-provider/action-plan/${actionPlanId}/review`)
+        try {
+          await this.interventionsService.updateDraftActionPlan(token, actionPlanId, form.paramsForUpdate)
+          return res.redirect(`/service-provider/action-plan/${actionPlanId}/review`)
+        } catch (e) {
+          if (e.status === 400) {
```

Comment:
How do we know that it will always be an issue with number of sessions?

Seems that we should have a check on the body returned containing `numberOfSessions` with error `CANNOT_BE_REDUCED`?

Reply:
Yeah, I was thinking the same - I've realised we already have a mechanism for handling server errors like this, so have re-used that functionality.